internal/websocket/ws.go

```diff
@@ -371,17 +371,42 @@ func (rt *custRT) RoundTrip(r *http.Request) (*http.Response, error) {
 func (ws *Websocket) updateClientTransport(mode ipMode) {
 	// Override client's Transport with custRT (reusing certain configurations)
 	// and update it's DialContext with the provided mode.
```

Comment:
when `updateClientTransport()` is called more than once, `ws.client.Transport.(*http.Transport)` will trigger the panic `http.RoundTripper is *websocket.custRT, not *http.Transport`